RecoTracker/TkTrackingRegions/interface/TrackingRegion.h

```diff
@@ -96,6 +99,14 @@ class TrackingRegion {
   /// get hits from layer compatible with region constraints
   virtual Hits hits(const edm::EventSetup& es, const SeedingLayerSetsHits::SeedingLayer& layer) const = 0;
 
+  /// updates an existing  mask over a track collection reflecting the compatability to the region
+  virtual void checkTracks(reco::TrackCollection const& InputCollection, std::vector<bool>& mask) const = 0;
```

Comment:
`InputCollection` should be lower case; if it doesn't cause conflicts, I would suggest to use `tracks` everywhere.